geonode_mapstore_client/client/js/api/geonode/v2/index.js

```diff
@@ -370,6 +370,12 @@ export const getResourceTypes = ({}, filterKey = 'resource-types') => {
         });
 };
 
+export const getLayerByName = name => {
+    const url = parseDevHostname(`${endpoints[LAYERS]}/?filter{alternate}=${name}`)
+    return axios.get(url)
+    .then(({data}) => data);
```

Comment:
We could return directly the layer here instead of the data because from the name of the function I would expect a layer object.
```js
 .then(({data}) => data?.layers[0]);
 ```

Reply:
With the new requirement we should replace this api to get multiple layer information with one request and we could use a structure like this:
```js
const getLayersByName = names => { // names is an array of layer names
    const url = parseDevHostname(endpoints[LAYERS])
    return axios.get(url, {
        params: {
            page_size: names.length,
            'filter{alternate.in}': names
        }
    })
    .then(({data}) => data?.layers);
```